docs/configuration.md

```diff
@@ -0,0 +1,214 @@
+# Configuration variables
+
+Arctic has several tuning knobs under [arctic/_config.py](https://github.com/manahl/arctic/blob/master/arctic/_config.py) which affect the functionality of certain modules.
+
+Most of these variables are initialized via environment variables, which are explained in the rest of this section.
+
+
+<br><br>
+
+## VersionStore
+
+### STRICT_WRITE_HANDLER_MATCH
+
+Controls if Arctic can only match the intended handlers for the data type. If set to true it prevents falling back to pickling if the matching handler (based on the type of data) can't serialize without objects.
+
+```
+export STRICT_WRITE_HANDLER_MATCH=1
+```
+
+
+
+<br><br>
+
+
+## NdArrayStore
+
+### CHECK_CORRUPTION_ON_APPEND
+
+Enables more thorough sanity checks for detecting data corruption when issuing appends. The checks will introduce a 5-7% performance hit. This is disabled by default.
+
+```
+export CHECK_CORRUPTION_ON_APPEND=1
+```
+
+
+
+<br><br>
+
+
+## Serialization
+
+### ARCTIC_AUTO_EXPAND_CHUNK_SIZE
+
+If a row is too large, then auto-expand the data chunk size from the default _CHUNK_SIZE (it is 2MB). It is disabled by default, and the written DataFrame in its serialized Numpy array form shouldn't exceed 2MB.
+
+This setting is effective only when using the incremental serializer.
+
+```
+export ARCTIC_AUTO_EXPAND_CHUNK_SIZE=1
+```
+
+
+### MAX_DOCUMENT_SIZE
+
+This configuration variable is used only when ARCTIC_AUTO_EXPAND_CHUNK_SIZE is set and the user writes a DataFrame which has extremely large number of columns (exceed 2MB serialized).
+
+This value must be less than 16MB which is the maximum document size of MongoDB, taking into account the size of other fields, serialized as a BSON object.
+
+Its default value is configured as follows:
```

Comment:
configured confused me here, maybe calculated?

Reply:
It is a code static variable, not configurable via os.env (on purpose).
In this context it seems better to convert to:

```
It is set to the following default value:
```